### PR TITLE
PIM-3661: Rework PIM-3661 PR to make it cleaner

### PR DESCRIPTION
**Pull request by solivier**
| Q                    | A
| -------------------- | ---
| Bug fix?             | No
| New feature?         | No
| BC breaks?           | Yes
| CI currently passes? | Yes
| Tests pass?          | Yes
| Scenarios pass?      | Running
| Checkstyle issues?*  | No
| PMD issues?**        | No
| Changelog updated?   | Yes
| Fixed tickets        | PIM-3661
| DB schema updated?   | No
| Migration script?    | No
| Doc PR               | No

### CODE REVIEW

File: src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @author    Olivier Soulet <olivier.soulet@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class VersionContext
+{
+    /**
+     * Versioning context
+     *
+     * @var array
```
Comment (nidup) [inline src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php]: phpdoc on one line plz 

/** @var array */

File: src/Pim/Bundle/VersioningBundle/Manager/VersionManager.php [hunk 2]
```diff
@@ -94,37 +101,6 @@ public function isRealTimeVersioning()
     }
 
     /**
-     * Set context
-     *
-     * @param string      $context
-     * @param string|null $fqcn
-     */
-    public function setContext($context, $fqcn = null)
```
Comment (nidup) [inline src/Pim/Bundle/VersioningBundle/Manager/VersionManager.php]: update the changelog please

File: src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        $this->context['default'] = null;
+    }
+
+    /**
+     * Set context
+     *
+     * @param string      $context
+     * @param string|null $fqcn
+     */
+    public function addContext($context, $fqcn = null)
```
Comment (nidup) [inline src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php]: addContext, getContext in a VersionContext class is weird,

File: src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Pim\Bundle\VersioningBundle\Manager;
+
+/**
+ * Version manager
```
Comment (nidup) [inline src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php]: Version context

File: src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php (hunk 1 above)
Comment (solivier) [inline src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php]: there is a description for the var

File: src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php (hunk 3 above)
Comment (jjanvier) [inline src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php]: addContextInfo($context, $info = null) ?

File: src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    /**
+     * Get context
+     *
+     * @param null $fqcn
+     *
+     * @return null|string
+     */
+    public function getContext($fqcn = null)
```
Comment (jjanvier) [inline src/Pim/Bundle/VersioningBundle/Manager/VersionContext.php]: addContextInfo($context) ?

File: CHANGELOG.md [hunk 6]
```diff
@@ -171,6 +171,8 @@
 - You cannot add product to multiple variant group anymore
 - constructor of `Pim\Bundle\CatalogBundle\Entity\Repository\GroupRepository` to add ProductTemplateUpdaterInterface and Validator interface
 - rename buildProductValueForm to createProductValueForm in `Pim\Bundle\EnrichBundle\Form\Factory\ProductValueFormFactory`
+- The method `setContext` for the class `src/Pim/Bundle/VersioningBundle/Manager/VersionManager` has been moved to `src/Pim/Bundle/VersioningBundle/Manager/VersionContext` and renamed setContextInfo
+- The method `getContext` for the class `src/Pim/Bundle/VersioningBundle/Manager/VersionManager` has been moved to `src/Pim/Bundle/VersioningBundle/Manager/VersionContext` and renamed getContextInfo
```
Comment (jjanvier) [inline CHANGELOG.md]: there a lot of others BC ^^

Comment (solivier) [inline CHANGELOG.md]: updated thanks

**Pull request merged**